text/0000-offline-buildpackages.md

```diff
@@ -0,0 +1,234 @@
+│           │   ├── build
+│           │   └── detect
+│           └── buildpack.toml
+├─ deps
+│   ├── io.paketo.mri
+│   │    ├── v2.7.1.toml
+│   │    ├── v2.7.1
+│   │    ├── v2.6.8.toml
+│   │    └── v2.6.8
+│   ├── io.paketo.other-dep
+│   │    ├── v2.2.2.toml
+│   │    └── v2.2.2
+│   └── <id>
+│        ├── v1.2.3.toml
+│        ├── v1.2.3
+│        ├── <version>
+│        └── <version>.toml
+└── lifecycle
+```
+### How will paths to dependencies be exposed to a buildpack?
+
+During the `build` phase of the `paketo-buildpacks/mri` buildpack, the `CNB_DEPS` environment variable will provide a path pointing to a `packaged.toml` file, this will contain information on all dependencies packaged with a buildpack.
+
+More on the contents of this file [here](#What-are-the-contents-of-the-packagedtoml-file-that-CNB_PATH-points-to).
+
+
+### How are dependencies added to a buildpackage, what is the `pack` user experience?
+These dependencies are added to a buildpackage by specifying fields in the `package.toml`
+
+``` toml
+[buildpack]
+uri = "./path/to/buildpack.tgz"
+
+[[dependencies]]
+uri = "path/to/dependency"
+id = "org.some-identifier"
+version = "v1.2.3"
+  [dependencies.metadata]
+  # dependency-specific data
+
+[[dependencies]]
+uri = "https://buildpacks.cloudfoundry.org/dependencies/node/node_10.20.0_linux_x64_cflinuxfs3_76f4a198.tgz"
+ID = "paketo.node"
+version = "v10.20.0"
+  [dependencies.metadata]
+  cool = "beans"
+
+
+```
+
+the above would give us the following file tree.
+
+```
+cnb
+└─ deps
+    ├── org.some-identifier
+    │    ├── v1.2.3.toml
+    │    └── v1.2.3
+    └── paketo.node
+         ├── v10.20.0.toml
+         └── v10.20.0
+
+```
+
+
+### How are dependencies grouped as layers
+Layers containing dependencies are attached as the final layers to a buildpackage. The exact distribution of dependencies into layers is left up to the packaging utility.
+
+
+### How are dependency layers matched with a buildpack?
+A buildpack packaged with dependencies will have the added `depLayers` field in the `org.buildpack.buildpack.layers` metadata.
+
+``` json 
+"paketo-buildpacks/node-engine":{
+      "0.0.178":{
+         "api":"0.2",
+         "stacks":[
+            {
+               "id":"org.cloudfoundry.stacks.cflinuxfs3"
+            },
+            {
+               "id":"io.buildpacks.stacks.bionic"
+            }
+         ],
+         "layerDiffID":"sha256:9ae53c280ea60b30b8c03978fbd2f761769fe0c15e421a9ece68e2f52926ce77",
+         "depLayers":{
+              "sha256:sha256-to-dep-layer": {
+                 "path": "/cnb/deps/path/to/dep1",
+                 "id": "org.some-identifier",
+                 "version": "v1.2.3", "metadata": {}
+              },
+              "sha256:sha256-to-other-dep-layer": {
+                 "path": "/cnb/deps/path/to/dep1",
+                 "id": "paketo.node",
+                 "version": "v10.20.0",
+                 "metadata": {"cool": "beans"}
+              }
```

Comment:
This format seems to preclude adding multiple deps in a single layer in the future. Should we make each `depLayer` contain an array of dependencies so that we have some breathing room if/when we need to design a strategy for dealing with the layer limit?

Reply:
Also, maybe we should pull the `depLayers` out the buildpack map,  in case multiple buildpacks want to share an identical dep layer?